library/std/src/path.rs

```diff
@@ -334,6 +334,29 @@ fn split_file_at_dot(file: &OsStr) -> (Option<&OsStr>, Option<&OsStr>) {
     }
 }
 
+fn split_file_at_dot(file: &OsStr) -> (Option<&OsStr>, Option<&OsStr>) {
```

Comment:
This doesn't seem to ever return None for the tuple's first value. Is that intentional?

```suggestion
fn split_file_at_dot(file: &OsStr) -> (&OsStr, Option<&OsStr>) {
```

Reply:
That is true. I just copied and tweaked the implementation for `rsplit_file_at_dot` which is used by `Path::file_stem`. Should I change the return signature as you suggest or keep it inline with `rsplit_file_at_dot`?

Reply:
Please change the return type per the suggestion.